core/src/main/scala/org/apache/spark/scheduler/CoarseCookSchedulerBackend.scala

```diff
@@ -444,11 +475,22 @@ class CoarseCookSchedulerBackend(
   override def start(): Unit = {
     super.start()
 
+    // We relay on client to set the `executorLimitOption` via calling `doRequestTotalExecutors`
+    // indirectly/directly. In the case of dynamic allocation, it is done when creating
+    // `SparkContext` and calling `ExecutorAllocationManager.start()`, otherwise, we need to
+    // set it explicitly.
+    if (!schedulerContext.isDynamicAllocationEnabled) {
+      doRequestTotalExecutors(
```

Comment:
can we make the unit that the backend deals in 'executors' rather than 'cores'?  I think this division belongs at a higher level.  At this point in the code we should just care about the number of executors we need.